hibernate-core/src/main/java/org/hibernate/event/internal/DefaultLoadEventListener.java

```diff
@@ -292,31 +292,12 @@ private Object proxyOrLoad(
 				}
 			}
 
+			// Potentially add a batch-fetch entry into the queue for this entity
+			persistenceContext.getBatchFetchQueue().addBatchLoadableEntityKey( keyToLoad );
```

Comment:
Minor nit-picking: invoking `getBatchFetchQueue()` will create a queue if there isn't one already. This object is quite large, so we should try to avoid it when not necessary.

It's possible that batching was not enabled, so we should check `keyToLoad.isBatchLoadable()` before doing this.

Reply:
I have this change locally, I guess I can just push your PR and then add a commit.

Reply:
ok my comment is unesefull, sorry for the noise

Reply:
So then every single call to this method should be changed to this paradigm as well.

Reply:
I wonder if a better option might be to delay creating the Maps inside `BatchFetchQueue`.  That way we do not have to make the same change everywhere this same current paradigm is used - and if we make the change in one place we need to make it in all imo.  The idea being that it is no longer as heavy to just create the `BatchFetchQueue`.

Reply:
The Travis failure is a test failure in `hibernate-osgi`.  Pretty sure that is not related to these changes :(

Reply:
Re-started the Travis build just to see if it was maybe (hopefully) a transient problem

Reply:
> I wonder if a better option might be to delay creating the Maps inside

I am just doing that.. will push soon.